loading.bs

```diff
@@ -331,6 +386,391 @@ Note: Applying the signed exchange's response here has the effect of letting a
+
+    9. If |sourceBrowsingContext|'s [=active document=]'s
+        [=Document/prefetched signed exchanges for navigation=][|request|'s
+        [=request/url=]] [=map/exists=], then:
+
+        1. Let |prefetchedExchange| be |sourceBrowsingContext|'s 
+            [=active document=]'s [=Document/prefetched signed exchanges for
+            navigation=][|request|'s [=request/url=]].
+
+        1. Set |request|'s [=request/stashed exchange=] to the result of
+            creating a new [=exchange=] with |prefetchedExchange|'s
+            [=exchange/request URL=] and the result of [=response/cloning=]
+            |prefetchedExchange|'s [=exchange/response=].
+
+
+And before
+
+> 17. Run process a navigate response with navigationType, the source browsing
+>     context, and navigationParams.
+
+add the following steps:
+
+    18. Copy |sourceBrowsingContext|'s [=active document=]'s
+        [=Document/prefetched subresource signed exchanges=]
+        to <var ignore=''>navigationParams</var>'s
+        [=navigation params/prefetched subresource signed exchanges=].
+
+Note: As browsers move toward partitioned HTTP caches, the source document's
+    cache will likely be separate from the target's cache, so we can't just pass
+    prefetched content through the cache.
+
+
+<h3 algorithm id="mp-page-load-processing-model-for-html-fiiles">
+Monkeypatch Page load processing model for HTML files</h3>
+
+Note: To use the prefetched subresource signed exchanges after the navigation,
+this section monkeypatchs [=Page load processing model for HTML files=].
+
+Issue(whatwg/html#4224): Currently the processing model of the HTTP <a
+http-header>Link</a> header is not defined in the HTML spec. This section
+monkeypatches only for <a http-header>Link</a> rel=preload HTTP headers. When we
+will change the HTML spec to generally support <a http-header>Link</a> HTTP
+headers, this section must be made align with the change.
+
+
+In [=Page load processing model for HTML files=] before
+
+> 2. Create an HTML parser and ...
+
+add the following steps:
+    2. Run the following steps [=in parallel=]:
+        1. Wait until |document|'s [=viewport=] is known.
+
+            Note: When the |document|'s [=viewport=] is known is not normatively
+            defined in the HTML spec. Need to define the behavior of
+            preloadScanner to define this.
+        1. Let |linkHeader| be the result of [=header list/getting=] `` `Link`
+            `` from |navigationParams|'s [=navigation params/response=]'s
+            [=response/header list=].
+        1. If |linkHeader| is null, then return.
+        1. Let |links| be the result of [=Parsing a Link Field Value=] from
+            |linkHeader|.
+        1. Let |allowedSxgLinks| be the result of [=getting allowed signed
+            exchange link info=] from |links|.
+        1. Let |preloadLinkItems| be an empty [=list=].
+        1. Let |canLoadAlternateSxg| be true. 
+        1. For each |link| of |links|:
```

Comment:
Non blocking note: Might be worthwhile to coalesce the common parts of the 2 large fetching algorithms into a single one, called from both places. As is, there's a lot of repetition.